typing/jkind.ml

```diff
@@ -1993,7 +2084,7 @@ module Jkind_desc = struct
       List.fold_right
         (fun (type_expr, modality) bounds ->
           With_bounds.add_modality ~relevant_for_nullability:`Relevant
-            ~type_expr ~modality bounds)
+            ~relevant_for_separability:`Irrelevant ~type_expr ~modality bounds)
```

Comment:
I'm surprised the two shallow axes are different. I suspect this is wrong.

Reply:
Separability for products is always non-float.